xdsl/interpreters/__init__.py

```diff
@@ -26,6 +26,7 @@ def register_implementations(
     interpreter: Interpreter,
     ctx: MLContext,
```

Comment:
```suggestion
    ctx: MLContext,
    *,
```
Now that we have multiple optional boolean parameters, I'd recommend adding a `*` parameter, to force the boolean flags to be named.